### DISCUSSION TITLE
How to order record of relationship fields?

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#1251] [Q&A (Help)]
I have created a relationship field and I need a way to reoder items like a repeater:

     CRUD::field([
            'name' => 'paths',
            'label' => __('backend/museum.info.paths'),
            'type' => 'relationship',
            'entity' => 'paths',
            'attribute' => 'name',
            'pivot' => true,
            'multiple' => true,
            'model' => 'App\Models\MuseumPath',
            'inline_create' => ['entity' => 'museum-path'],
            'wrapper' => [
                'class' => 'form-group col-md-12'
            ],
            'subfields' => [],
            'pivotSelect' => [
                'label' =>  __('backend/museum.paths.select_path'),
                'inline_create' => [
                    'entity' => 'museum-path',
                    'modal_route' => route('museum-path-inline-create'),
                ],
                'data_source' => route('museum.fetchMuseumPaths'),
            ],
            'tab' => $paths_tab,
        ]);

how can I do that?

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment]
Hey @cod3rshotout

The bot is partially correct. Follow these steps to reorder the `subfield` rows of the `relationship` field:

1. You need to add an 'order' column to your pivot table if it doesn't already exist.
2. Make sure to add the 'order' column to your `withPivot()` statement in your relationship definition.
```php
public function owners()
{
    return $this->belongsToMany(\App\Models\PetShop\Owner::class)->withPivot(['role', 'order']);
}
```
3. Now add the `reorder` attribute to the field definition
```diff
CRUD::field('owners')
  ->subfields([['name' => 'role', 'type' => 'text']])
+  ->reorder('order');
```

This will add up and down arrows next to each row in the relationship field, allowing you to reorder the items. You can also check a diff [here](https://github.com/Laravel-Backpack/demo/compare/main...test-order-column-attr) of a branch I used to test before answering.

**Reply by karandatwani92** [reply]
The docs had this `reorder` attribute in the `repeatable` field definition.
I'll update the docs with the same for the 'relationship' field for better clarity.

Thanks for pointing out and opening the discussion.
